pmlc/rt/vulkan/vulkan_runtime.cc

```diff
@@ -830,12 +839,46 @@ LogicalResult VulkanRuntime::createCommandPool() {
   commandPoolCreateInfo.pNext = nullptr;
   commandPoolCreateInfo.flags = 0;
   commandPoolCreateInfo.queueFamilyIndex = queueFamilyIndex;
+  llvm::errs() << "queueFamilyIndex = " << queueFamilyIndex;
   RETURN_ON_VULKAN_ERROR(
       vkCreateCommandPool(device, &commandPoolCreateInfo, 0, &commandPool),
       "vkCreateCommandPool");
   return success();
 }
 
+LogicalResult VulkanRuntime::createTimestampQueryPool() {
+  /*
+  typedef struct VkQueryPoolCreateInfo {
+    VkStructureType                  sType;
+    const void*                      pNext;
+    VkQueryPoolCreateFlags           flags;
+    VkQueryType                      queryType;
+    uint32_t                         queryCount;
+    VkQueryPipelineStatisticFlags    pipelineStatistics;
+  } VkQueryPoolCreateInfo;
+  */
+
+  VkQueryPoolCreateInfo queryPoolCreateInfo = {};
+  queryPoolCreateInfo.sType = VK_STRUCTURE_TYPE_QUERY_POOL_CREATE_INFO;
+  queryPoolCreateInfo.pNext = nullptr;
+  queryPoolCreateInfo.flags = 0;
+  queryPoolCreateInfo.queryType = VK_QUERY_TYPE_TIMESTAMP;
+  queryPoolCreateInfo.queryCount = 1024; // TODO: hardcode?
```

Comment:
I think is OK. We probably will need a larger number if we need to track every single gpu module.